shell/platform/windows/flutter_windows_view.cc

```diff
@@ -255,7 +286,26 @@ bool FlutterWindowsView::ClearContext() {
 }
 
 bool FlutterWindowsView::SwapBuffers() {
-  return surface_manager_->SwapBuffers();
+  std::unique_lock<std::mutex> lock(resize_mutex_);
+
+  switch (resize_status_) {
+    // SwapBuffer requests during resize are ignored until the frame with the
+    // right dimensions has been generated. This is marked with
+    // kFrameGenerated resize status.
+    case ResizeState::kResizeStarted:
+      return false;
+    case ResizeState::kFrameGenerated: {
+      bool swap_buffers_result = surface_manager_->SwapBuffers();
+      resize_status_ = ResizeState::kDone;
+      resize_cv_.notify_all();
+      lock.unlock();
```

Comment:
std::condition_variable docs point out that one should unlock before, rather than after, notifying, to avoid waking up a thread only to have it immediately be blocked on a lock.

Reply:
Good catch!